core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1446,6 +1480,12 @@ public TableMetadata build() {
           "Cannot set metadata location with changes to table metadata: %s changes",
           changes.size());
 
+      if (hasRemovedSpecs) {
+        Preconditions.checkArgument(
+            !hasPartitionSpecUpdates(),
+            "Cannot remove partition specs with other partition spec update");
```

Comment:
This should not be a requirement. It is reasonable to add a new spec, set the default partition spec, and then remove the old spec.